include/pmacc/lockstep/Kernel.hpp

```diff
@@ -67,65 +67,185 @@ namespace pmacc::lockstep
              * Object is used to apply the grid and block extents and optionally the amount of dynamic shared memory to
              * the kernel.
              */
-            template<typename T_KernelFunctor, typename T_WorkerCfg>
+            template<typename T_UserKernelFunctor>
             struct KernelPreperationWrapper
             {
-                using KernelFunctor = detail::LockStepKernel<T_KernelFunctor, T_WorkerCfg>;
-                KernelFunctor const m_kernelFunctor;
+                template<typename T_BlockCfg>
+                using KernelFunctor = detail::LockStepKernel<T_UserKernelFunctor, T_BlockCfg>;
+
+                T_UserKernelFunctor const m_UserKernelFunctor;
                 pmacc::exec::detail::KernelMetaData const m_metaData;
 
                 HINLINE KernelPreperationWrapper(
-                    T_KernelFunctor const& kernelFunctor,
+                    T_UserKernelFunctor const& kernelFunctor,
                     std::string const& file = std::string(),
                     size_t const line = 0)
-                    : m_kernelFunctor(kernelFunctor)
+                    : m_UserKernelFunctor(kernelFunctor)
                     , m_metaData(file, line)
                 {
                 }
 
                 /** Configured kernel object.
-                 *
-                 * This objects contains the functor and the starting parameter.
                  *
                  * @tparam T_VectorGrid type which defines the grid extents
-                 * @tparam T_VectorBlock type which defines the block extents
+                 * @tparam T_blockSize number of block indices
                  *
-                 * @param gridExtent grid extent configuration for the kernel
-                 * @param blockExtent block extent configuration for the kernel
+                 * @param gridSize grid extent configuration for the kernel
                  *
                  * @return object to bind arguments to a kernel
                  *
                  * @{
                  */
+                template<uint32_t T_blockSize, typename T_VectorGrid>
+                HINLINE auto config(T_VectorGrid const& gridSize) const
```

Comment:
What was the thought behind changing from operator()() to config()

Reply:
it became more human readable and is required to allow to pass the block size as scalar value `.config<256>()`. This is not possible with the functor oeprator.